### PR TITLE
Extend sdk api to retrieve raw config

### PR DESCRIPTION
**Pull request by alexsavu**
This added functionality allows for different use cases of the current config such as event tracking, logging etc. This will allow for direct retrieval of latest config stored locally.

* extend Environment to allow apis to run on local server through http
* tests for added api

### CODE REVIEW

File: sdk/ios/Configsum/Configsum.swift [hunk 1]
```diff
@@ -95,4 +95,11 @@ public struct Configsum {
                                       success: success,
                                       failure: failure)
     }
+    
+    /// Returns the entire stored config
+    ///
+    /// - Returns: th   e config in the form for a key-value data structure
```
Comment (mictab) [inline sdk/ios/Configsum/Configsum.swift]: Something went wrong here with the formatting.

File: sdk/ios/Configsum/Configuration/Environment.swift [hunk 2]
```diff
@@ -23,18 +24,21 @@ public struct Environment {
...
     ///   - port: optional attribute for http/https requests
+    ///   - urlScheme: The scheme subcomponent of the URL. if not set it defaults to https
     public init(log: Bool,
                 token: String,
                 headers: [String: [String]],
                 baseConfigurationName: String,
                 hostName: String,
-                port: Int? = nil) {
+                port: Int? = nil,
+                urlScheme: String = "https") {
```
Comment (mictab) [inline sdk/ios/Configsum/Configuration/Environment.swift]: What happens if the caller provides an unsupported scheme? Should we make this an enum instead maybe?

File: sdk/ios/ConfigsumTests/ContextTests.swift [hunk 3]
```diff
@@ -91,6 +91,8 @@ class ContextTests: XCTestCase {
         
         let resultString = String(data: try! encoder.encode(context), encoding: .utf8)!
         let expectedString = String(data: try! encoder.encode(wantedContext), encoding: .utf8)!
+        print("RESULT: \(resultString)")
```
Comment (mictab) [inline sdk/ios/ConfigsumTests/ContextTests.swift]: I guess these prints aren't needed anymore?

File: sdk/ios/ConfigsumTests/contextWithComplexMetadata.json [hunk 4]
```diff
@@ -18,7 +18,7 @@
         },
         "location" : {
             "locale" : "en-US",
-            "timezoneOffset" : 7200
+            "timezoneOffset" : 3600
```
Comment (mictab) [inline sdk/ios/ConfigsumTests/contextWithComplexMetadata.json]: Any reason these were changed? :)

File: sdk/ios/Configsum/Configuration/Environment.swift (hunk 2 above)
Comment (mictab) [reply sdk/ios/Configsum/Configuration/Environment.swift]: From the Apple docs:

`If you attempt to set the scheme to an invalid scheme string, this class throws an exception.`

File: sdk/ios/ConfigsumTests/contextWithComplexMetadata.json (hunk 4 above)
Comment (alexsavu) [reply sdk/ios/ConfigsumTests/contextWithComplexMetadata.json]: code for calculating timezoneOffset was added

File: sdk/ios/Configsum/Configuration/Environment.swift (hunk 2 above)
Comment (alexsavu) [reply sdk/ios/Configsum/Configuration/Environment.swift]: introducing a enum will break older clients

**Pull request merged**